internal/commonbuilder/commonbuilder_test.go

```diff
@@ -0,0 +1,100 @@
+package commonbuilder
+
+import (
+	"testing"
+
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	siteconfigv1alpha1 "github.com/openshift-kni/eco-goinfra/pkg/schemes/siteconfig/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/stretchr/testify/assert"
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestCommonBuilderValidate(t *testing.T) {
+	testCases := []struct {
+		builderNil    bool
+		definitionNil bool
+		apiClientNil  bool
+		expectedError string
+	}{
+		{
+			definitionNil: true,
+			apiClientNil:  false,
+			expectedError: "can not redefine the undefined clusterinstance",
+		},
+		{
+			definitionNil: false,
+			apiClientNil:  true,
+			expectedError: "clusterinstance builder cannot have nil apiClient",
+		},
+		{
+			definitionNil: false,
+			apiClientNil:  false,
+			expectedError: "",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testBuilder := generateTestBuilder()
+
+		if testCase.definitionNil {
+			testBuilder.Definition = nil
+		}
+
+		if testCase.apiClientNil {
+			testBuilder.apiClient = nil
+		}
+
+		testCommonBuilder := New(testBuilder)
+
+		result, err := testCommonBuilder.Validate()
+		if testCase.expectedError != "" {
+			assert.NotNil(t, err)
+			assert.Equal(t, testCase.expectedError, err.Error())
+			assert.False(t, result)
+		} else {
+			assert.Nil(t, err)
+			assert.True(t, result)
+		}
+	}
+}
+
+func generateTestBuilder() *testBuilder {
+	return &testBuilder{
+		apiClient: clients.GetTestClients(
+			clients.TestClientParams{}).Client,
+		Definition: generateClusterInstance(),
```

Comment:
Used clusterinstance for this test so I wouldn't need to create my own struct that implemented everything required by runtime objects. The idea is that this shouldn't matter though, you could replace it with any pkg using runtime objects.